docs/developer_guide.md

```diff
@@ -1,11 +1,47 @@
 # Developer Guide
 Thank you for contributing to RepoSense!
+- [Setting Up](#setting-up)
+- [Architecture](#architecture)
+- [Notable POJOs](#notable-pojos)
+- [Frontend Components](#frontend-components)
+- [Dashboard Generator Components](#dashboard-generator-components)
+- [Git Utilities](#git-utilities)
+- [Analyzer Components](#analyzer-components)
+- [HTML Dashboard](#html-dashboard)
+
+## Setting up
+
+### Prerequisites
+
+1. **JDK `1.8.0_60`**  or later<br>
+> This app only works with Java 8 version. <br>
+
+2. **IntelliJ** IDE
```

Comment:
Why are we restricting IDE to IntelliJ

Reply:
hmm, restricting is not my intention but a common IDE will probably be easier for multiple developers to work together. IntelliJ is the dedicated IDE for java in the jetbrains suite of tools. 
I'm open to suggestions though.

Reply:
In the context of this project, I cannot think of any specific advantages for unifying IDE... But I think it will be good for beginners! 
The other two requirements are JDK and Git, which are both compulsory, so it might imply that IntelliJ is compilsory too. It might cause some confusion.
Maybe mark it as optional to avoid confusion? Thx

Reply:
Yes, we can make Intellij an example only.

Reply:
> In the context of this project, I cannot think of any specific advantages for unifying IDE... But I think it will be good for beginners!
The other two requirements are JDK and Git, which are both compulsory, so it might imply that IntelliJ is compilsory too. It might cause some confusion.
Maybe mark it as optional to avoid confusion? Thx

Alright